README.md

```diff
@@ -7,10 +7,137 @@ new [cisagov](https://github.com/cisagov) Go project started.
+version number within the current directory.
+
+Execute the `./skeleton-golang-package-v*` binary to run the program.
+
+You should see the contents of the main package as defined in [main.go](main.go).
+
+### Nix ###
+
+If you have [Nix](https://nixos.org/download.html) installed you can use
+the [flake.nix](flake.nix) configuration file located at the root of the
+project to build and run the `skeleton-golang-package` package:
+
+```console
+nix run
+```
+
+After the build has completed the program will automatically execute. The build
+will also have created a `result` directory within the root of the project.
+This directory contains a `bin` subdirectory where the executable
+`skeleton-golang-package` can be found:
+`result/bin/skeleton-golang-package`
+
+To run the program simply execute the binary from the project root:
+
+```console
+result/bin/skeleton-golang-package
+```
+
+Alternatively, if you want to execute an interactive build environment
+with all available dependencies for this project you can run this
+command in the project root:
+
+```console
+nix develop
+```
+
+If you make changes to the project files you will need to update `vendorSha256`
+in `flake.nix`  You can do this by running `nix build` and you should be presented
+with an error similar to:
+
+```console
+error: hash mismatch in fixed-output derivation '/nix/store/11a71sp1wynjgxinx6yb0yhli4q659zi-skeleton-golang-package-20230424-go-modules.drv':
+         specified: sha256-+aBatR0sHWkykrQr8AMD0P5xoc4VEXtA+egp6PxsKzY=
+            got:    sha256-gqfvjULp2VApWQl7yFVj45meYpcS4XefUtEUy+TtAH4=
+error: 1 dependencies of derivation '/nix/store/nm6xhgfawf67sy89rl9azdzvnjc5r7cr-skeleton-golang-package-20230424.drv' failed to build
+```
+
+You would then copy the `got` hash and use it to replace the previous
+ `vendorSha256` value within `flake.nix`.
+
+If `flake.nix` changes or is modified you will also need to update the
+`flake.lock` file. You can do this by running: `nix flake update`. It is
+also recommended to then run `nix build` after this operation to ensure
+that the build is successful.
+
+## SBOM ##
+
+A Software Bill of Materials (SBOM) is like a recipe card for software,
+detailing all the ingredients (components, libraries, dependencies)
+used to cook up a digital product. It's a crucial tool for understanding
+what goes into software, helping with things like spotting vulnerabilities,
+ensuring license compliance, and keeping an eye on supply chain security.
+However, SBOM isn't a magic fix for security issues itself—it's more of a
+roadmap that helps teams navigate potential risks. It's not a substitute
+for good coding practices or regular updates, but a helpful companion in
+the journey towards stronger software security.
+
+The SBOM found in this project was generated using [cyclonedx-gomod](https://github.com/CycloneDX/cyclonedx-gomod).
+The exact commands used can be found below.
```

Comment:
@cisagov/team-ois Should we have the SBOM files generated as part of the automated build process?

Do these SBOM changes belong in skeleton-generic or any of the other skeleton repos? 

@felddy, I know you have spent time thinking about SBOM, so please add any thoughts you have here.

Reply:
@felddy thanks. so, in that case I will remove it since we're not pushing binaries.

Reply:
@jasonodoom Are you planning to do this in this initial PR?

> SBOMs related to a build artifact (binary, image, etc...) should be attached to the build process, and published as assets along side the release.

Reply:
I remember a discussion earlier in this PR where we said that we don't want to check binaries into source control, but in [this conversation](https://github.com/cisagov/skeleton-golang-package/pull/2#discussion_r1524318200), we are saying that we want to attach the built package as an artifact to the workflow run (e.g. like we do [here](https://github.com/cisagov/skeleton-python-library/actions/runs/8703681302)).  So if we do that, we should also add SBOMs for each artifact as well.  As with that other conversation, let us know if that is something that you plan to do in this PR or document in an issue as later work.

Anyone else, feel free to correct if I'm misunderstanding any of this.

Reply:
@dav3r I added additional info here too: https://github.com/cisagov/skeleton-golang-package/issues/13

Reply:
Thanks for including the SBOM artifact creation in #13. 👍